docs/docs/contributing/testing.md

```diff
@@ -54,6 +54,54 @@ You can also run all paths of classes of a certain name:
 > testOnly *.TreeTransformerTest
 ```
 
+### Testing with checkfiles
+Some tests support checking the output of the run or the compilation against a checkfile. A checkfile is a file in which the expected output of the compilation or run is defined. A test against a checkfile fails if the actual output mismatches the expected output.
+
+Currently, the `neg` (compilation must fail for the test to succeed) and `run` support the checkfiles. `neg`'s checkfiles contain an expected error output during compilation. `run`'s checkfiles contain an expected run output of the successfully compiled program.
```

Comment:
I would turn this sentence around. `run` check files are usually more important. Mention `run` before `neg`.

Reply:
Fixed